### PR TITLE
Example token auth

### PR DESCRIPTION
**Pull request by z-t-y**
Finishes the first item in #262 

Checklist:

- [ ] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the `docs` folder and in code docstring.
- [ ] Add an entry in `CHANGES.md` summarizing the change and linking to the issue and your username.
- [ ] Add `*Version changed*` or `*Version added*` note in any relevant docs and docstring.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

**Comment by greyli** [comment]
Thanks for working on this! I will review it later.

File: examples/auth/token_auth/app.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+@auth.verify_token
+def verify_token(token: str) -> t.Union[User, None]:
+    try:
+        data = jwt.decode(
+            token.encode('ascii'),
+            current_app.config['SECRET_KEY'],
+        )
+        id = data['id']
+    except BaseException:
```
Comment (greyli) [inline examples/auth/token_auth/app.py:45]: Using `authlib.jose.JoseError` will be more appropriate.

File: examples/auth/token_auth/app.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        return None
+    user = get_user_by_id(id)
+    g.current_user = user
```
Comment (greyli) [inline examples/auth/token_auth/app.py:48]: Just return the user, then get the current user in the view function with `auth.current_user`.

File: requirements/examples.txt [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
+    # via cryptography
 click==8.1.3
     # via flask
+cryptography==37.0.2
+    # via authlib
 flask==2.1.2
     # via flask-sqlalchemy
 flask-sqlalchemy==2.5.1
     # via -r requirements/examples.in
-importlib-metadata==4.11.3
```
Comment (greyli) [inline requirements/examples.txt:13]: Keep this item and the `zipp` for users with Python < 3.8.

File: examples/auth/token_auth/app.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        abort(404)
+    return {
+        'token': f'Bearer {get_user_by_id(id).get_token()}'
+    }
+
+
+@app.get('/name/<int:id>')
+@app.auth_required(auth)
+def get_secret():
+    if isinstance(g.current_user, User):
```
Comment (greyli) [inline examples/auth/token_auth/app.py:64]: This if statement seems unnecessary (?).

File: examples/auth/token_auth/app.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        id = data['id']
+    except BaseException:
+        return None
+    user = get_user_by_id(id)
+    g.current_user = user
+    return user
+
+
+@app.post('/token/<int:id>')
+def get_token(id: int):
```
Comment (greyli) [inline examples/auth/token_auth/app.py:53]: It's better to set up a `TokenSchema` with `app.output`.

**Review by greyli — CHANGES_REQUESTED** [review]

**Comment by z-t-y** [comment]
@greyli Changes have been made according to the reviews.

**Review by greyli — APPROVED** [review]

**Comment by greyli** [comment]
Thanks!

**Pull request merged**